protocol/deprecate-reqresp-sync.md

```diff
@@ -0,0 +1,138 @@
+# Deprecate request-response CL sync
+
+# Summary
+
+Deprecate CL-sync (request-response reverse-blocks sync, not block gossip)
+in favor of EL-sync, with improved sync-target reliability.
+
+This simplifies the op-node configuration, removes complicated sync-code,
+and makes the op-node more robust against P2P-network interruption.
+
+# Problem Statement + Context
+
+Currently the op-node has a "request-response" P2P sub-system to fetch blocks,
+verify them in reverse order against a sync target, and apply them to the EL node.
+This sub-system is complicated and fragile, and has existing better alternatives.
+
+## History
+
+### gossip-only sync
+
+The original op-node implementation supported GossipSub block distribution only.
+And the P2P functionality of the EL was unused.
+
+### req-resp reverse-sync
+
+One of the last features to land before Bedrock launch was reverse-sync,
+using the `request-response` protocol functionality of LibP2P.
+
+Since block-signatures are not stored, blocks can only be authenticated by their block-header parent-hash relation.
+This means that the op-node has to fetch blocks in reverse order: the blocks cannot immediately be processed.
+Parallelizing the fetching of these unauthenticated blocks, and then resolving the canonical chain, is complicated.
+And sync-depth is limited, due to memory limitations (the idea was to not introduce a database for just for syncing).
+
```

Comment:
I think it's worth linking to the issue that discusses the issues with the current impl: https://github.com/ethereum-optimism/optimism/issues/11779